src/builder/dataflow.rs

```diff
@@ -363,4 +365,74 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn lift_node() -> Result<(), BuildError> {
+        let mut module_builder = ModuleBuilder::new();
+
+        let ab_resources = ResourceSet::from_iter(["A".into(), "B".into()]);
+        let c_resources = ResourceSet::singleton(&"C".into());
+        let abc_resources = ab_resources.clone().union(&c_resources);
+
+        let mut parent_sig = Signature::new_df(type_row![BIT], type_row![BIT]);
+        parent_sig.output_resources = abc_resources.clone();
+        let mut parent = module_builder.define_function("parent", parent_sig)?;
```

Comment:
Note that you can start the Hugr with a DFG (of arbitrary signature), skipping the ModuleBuilder altogether. See https://github.com/CQCL-DEV/hugr/pull/234 for example; I think here you'd want `DFGBuilder::new`